### PR TITLE
erc20-lesson-update. Create intro structure (ready to review)

### PR DESCRIPTION
**Pull request by elPiablo**
ERC20 - My Token lesson 
 

We've decided to keep the PRs a bit more bite-size for the next round of content reviewing so we can get them live on the Academy platform.

This first of the new PRs for ERC20 - My Token lesson lets us add a meaningful introduction to the tutorial for the learner with the following list of components. 
(Future PRs for this lesson upgrade will allow for mini-quizzes and eventually a final quiz.)

~~- [ ] An "SEO approved" and meaningful title~~

- [x] Intro paragraph to clarify assumptions, and build confidence
  - Introduce learners' capabilities, competencies and experience, including proficiency with developer tools.
  - Clearly state intended learning outcome, referring to relevant previous lessons and setting context for subsequent lessons.
  - Offer estimated minimum and maximum completion times for lesson, catering to learners' respective competencies.
  - Learning extends far beyond the 'what' and 'how'. Provide the essential 'why', by showcasing real-world examples of how new knowledge can be used for social-environmental impact.

- [x] 2nd paragraph to introduce our pedagogy and lesson structure
  - We assess prior knowledge and test learning through mini-quiz checkpoints.
  - We engage learners with prediction activities and a final quiz. Guide them to join the forum.
  - We embrace well-being by encouraging regular breaks.
- [x] List of contents and relevant tools.

- [x] A lead-in to lesson.

- [ ] Brief description of upcoming ~~lesson~~ steps to consolidate the learner’s path e.g. link to Discord hackathon. Get a mentor, link to mentorship. Contribute to a DAO project and earn $CODE

~~- [ ] Kudos to learner, and prompt to engage in forum with peers.~~

### CODE REVIEW

**Comment by Markkos89** [comment]
git checkout erc20-lesson-update

git pull origin main `this will make the updates from v1.5 going to erc20-lesson-update branch`

git push origin erc20-lesson-update

**Comment by elPiablo** [comment]
@7i7o or @wolovim 
### A brief summary of changes to the lesson:
1 - I implemented the intro paragraphs structure that we are now using. 
2 - There were quite a lot of line breaks, so I joined up a lot of those sentences for 2 reasons:
a) most of the weight of text was coming from the left of the page, and it felt like reading a very long list of **sentences** of concepts.
b) I thought it would read more like a **story** of concepts, if it actually looked like a story. Personally, I thought it captured your traditional story telling better in paragraph form than in that list. That meant I just needed to add a few words here and there to join those sentences together.
3 - I moved a couple of sentences and/or concepts to different places to try to fit it into the paragraph structure in 1.
4 - I added a warm up quiz - more as a placeholder. If you get any good ideas for questions, it would be great to improve on them.

### @7i7o @wolovim If you are happy with the changes, I would say this is solid enough to merge for now, i.e. no real loose ends.

5 - next steps - @7i7o - you remember the convo we had about sections and sub-sections and calling them in JavaScript? - I'm a bit worried about the naming of the lesson sections as they are right now i.e. foundations, structure, doors, etc. for two reasons:
a) We are not telling the learner beforehand what to expect in the lesson (from a learning perspective)
b) I don't know how we can have sub-headers under these headers i.e. sections and sub-sections
I love the sense of surprise you by choosing this playful style of word choice. I have an idea around it, but it's a middle ground choice that kills off that element of surprise, so I would reserve it as a last resort. So I'd rather hear your usual creativity shine. Just so you know where I'm coming from, or would like to get to, is having:

Lesson steps
Dev tooling used in lesson
6 - next steps - follow up PRs with (mini-) quizzes and outro.
7 - next steps - any ideas for yourself, or others on a follow-up (series of) lesson(s). Or how do you see it?

**Comment by elPiablo** [comment]
> @7i7o or @wolovim
> 
> ### Request for review. A brief summary of changes to the lesson:
> 1 - I implemented the intro paragraphs structure that we are now using. 2 - There were quite a lot of line breaks, so I joined up a lot of those sentences for 2 reasons: a) most of the weight of text was coming from the left of the page, and it felt like reading a very long list of **sentences** of concepts. b) I thought it would read more like a **story** of concepts, if it actually looked like a story. Personally, I thought it captured your traditional story telling better in paragraph form than in that list. That meant I just needed to add a few words here and there to join those sentences together. 3 - I moved a couple of sentences and/or concepts to different places to try to fit it into the paragraph structure in 1. 4 - I added a warm up quiz - more as a placeholder. If you get any good ideas for questions, it would be great to improve on them.
> 
> @7i7o @wolovim
> 
> ### If you are happy with the changes, I would say this is solid enough to merge for now, i.e. no real loose ends.
> 5 - next steps - @7i7o - you remember the convo we had about sections and sub-sections and calling them in JavaScript? - I'm a bit worried about the naming of the lesson sections as they are right now i.e. foundations, structure, doors, etc. for two reasons: a) We are not telling the learner beforehand what to expect in the lesson (from a learning perspective) b) I don't know how we can have sub-headers under these headers i.e. sections and sub-sections I love the sense of surprise you by choosing this playful style of word choice. I have an idea around it, but it's a middle ground choice that kills off that element of surprise, so I would reserve it as a last resort. So I'd rather hear your usual creativity shine. Just so you know where I'm coming from, or would like to get to, is having:
> 
> Lesson steps Dev tooling used in lesson 6 - next steps - follow up PRs with (mini-) quizzes and outro. 7 - next steps - any ideas for yourself, or others on a follow-up (series of) lesson(s). Or how do you see it?

@wolovim @7i7o could one of you give this a once over, please?

**Comment by elPiablo** [comment]
@wolovim or @7i7o could you check and approve this commit?

**Review by elPiablo — COMMENTED** [review]
Hey @wolovim can you ignore the placement  of the CHECKPOINT QUESTIONS headers in this commit. I defined their placement more accurately/meaningfully in a later commit. Sorry for any  confusion.

File: src/pages/lessons/projects/6.mdx [hunk 1]
```diff
@@ -131,32 +146,32 @@ mkdir d_d_academy
 cd d_d_academy
 ```
 
-You can use whatever tools you feel confident or accostumed to. For this example
-we will be creating a Foundry project, but feel free to use Hardhat or Truffle
+You can use whatever tools you feel confident with, or accustumed to. For this example
```
Comment (wolovim) [inline src/pages/lessons/projects/6.mdx:149]: nearly there: accustomed* 😅

**Review by wolovim — COMMENTED** [review]
looks like a nice iteration 🍻 lets make sure to finish or remove the unfinished quiz links before this gets merged in.

File: src/utils/questions/lesson-6/1-intro/Q3.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      "answer": "Supporting a project"
+    },
+    {
+      "answer": "Governance for DAOs"
+    },
+    {
+      "answer": "Learning and experimentation"
+    },
+    {
+      "answer": "All of the above",
```
[the remark this answers is not in the document]
Comment (elPiablo) [reply src/utils/questions/lesson-6/1-intro/Q3.json:14]: Mmmmm, good consideration

File: src/utils/questions/lesson-6/1-intro/Q2.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  "options": [
+    {
+      "answer": "ERC-721",
+      "correct": true
+    },
+    {
+      "answer": "ERC-20"
+    },
+    {
+      "answer": "ERC-1155"
```
[the remark this answers is not in the document]
Comment (elPiablo) [reply src/utils/questions/lesson-6/1-intro/Q2.json:12]: yes :)

File: src/pages/lessons/projects/6.mdx [hunk 4]
```diff
@@ -19,43 +19,66 @@ import { LessonHeader } from "../../../components/mdx/LessonHeader";
   title="Create your own Tokens using Foundry"
 />
 
-## What are we creating?
+## About this lesson
+
+If you're eager to learn how to create your first ERC20 token, you're in the right place. There are
```
[the remark this answers is not in the document]
Comment (elPiablo) [reply src/pages/lessons/projects/6.mdx:24]: Appreciate these points

[the remark this answers is not in the document]
Comment (elPiablo) [reply src/pages/lessons/projects/6.mdx:24]: done :)

File: src/pages/lessons/projects/6.mdx (hunk 1 above)
Comment (elPiablo) [reply src/pages/lessons/projects/6.mdx:149]: or accostumed!! 😅😅😅

File: src/utils/questions/lesson-6/1-intro/Q2.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (elPiablo) [reply src/utils/questions/lesson-6/1-intro/Q2.json:12]: I like ERC-1559 - it has a double 5, and it's pretty well known, so if it trips someone up, it would hopefully do it in a kind way.

File: src/utils/questions/lesson-6/1-intro/Q3.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (elPiablo) [reply src/utils/questions/lesson-6/1-intro/Q3.json:14]: done :)

**Comment by elPiablo** [comment]
> looks like a nice iteration 🍻 lets make sure to finish or remove the unfinished quiz links before this gets merged in.

- Hey Marc, I just commented out the quiz links. Is that okay for now?

- As a bit of practice, I already re-ran the ERC-20 Foundry project afresh and created a separate branch locally with all the changes, including the resulting terminal outputs, any external URLs, doing the new screenshots for the assets,  etc. So that is ready to go.

- I have all the questions pretty much ready to go for the checkpoints and the final quiz. Just need to slot them into json format - all easy thanks to @0xChin !!!! Go Chiin!!

- And I had a couple of ideas for the outro (since we don't yet have any other lessons in this track) e.g. leading on to Intro to Eth, and the NFT track and another was using the https://handbook.developerdao.com/ as a follow up plan. 

Bottom line? Should we merge what we have, or do you want me to keep building, and how should we go about it? check your DMs :) https://discord.com/channels/@me/955461286416765018/1175611580327669820

**Review by wolovim — APPROVED** [review]
